src/Fcf/Data/Symbol.hs

```diff
@@ -0,0 +1,244 @@
+{-# LANGUAGE DataKinds              #-}
+{-# LANGUAGE PolyKinds              #-}
+{-# LANGUAGE TypeFamilies           #-}
+{-# LANGUAGE TypeInType             #-}
+{-# LANGUAGE TypeOperators          #-}
+{-# LANGUAGE UndecidableInstances   #-}
+{-# OPTIONS_GHC -Wall                       #-}
+{-# OPTIONS_GHC -Werror=incomplete-patterns #-}
+
+{-|
+Module      : Fcf.Data.Symbol
+Description : Type level symbols
+
+= Fcf.Data.Symbol
+
+Type-level symbols and functions for them.
+
+Note that the operators from this module conflict with "GHC.TypeLits".
+
+-}
+
+--------------------------------------------------------------------------------
+
+module Fcf.Data.Symbol
+    ( -- * Reexported type
+      -- | From "GHC.TypeList".
+
+      Symbol
+
+      -- * Functions
+
+    , Append
+    , Intercalate
+    , IsSpace
+    , IsNewLine
+    , IsTab
+    , IsSpaceDelim
+    , IsDigit
+
+     -- * Comparison functions
+
+    , SymbolOrd
+    , type (<=)
+    , type (>=)
+    , type (<)
+    , type (>)
+    , type (==)
+    )
+  where
+
+--------------------------------------------------------------------------------
+        
+import           GHC.TypeLits (Symbol)
+import qualified GHC.TypeLits as TL
+
+import           Fcf.Core (Eval, Exp)
+import           Fcf.Data.List (Foldr, Elem)
+import           Fcf.Data.Bool (type (||))
+import           Fcf.Utils (TyEq)
+
+import           Fcf.Combinators (type (=<<))
+
+--------------------------------------------------------------------------------
+
+-- | Append two type-level symbols.
+--
+-- === __Example__
+-- 
+-- >>> :kind! Eval (Append "hmm" " ok")
+-- Eval (Append "hmm" " ok") :: Symbol
+-- = "hmm ok"
```

Comment:
I prefer a more fun example, like "pineapple".

Reply:
yeah, much better that way, changed now